### DISCUSSION TITLE
Related column in Studio with custom `IdentityPartsSeparator`

### DISCUSSION THREAD

**Discussion by MikeMoolenaar99** [ravendb/ravendb#13379] [Q&A]
Studio doesn't show related documents. I do have `IdentityPartsSeparator` set to `-` in my C# code and the Raven client configuration. The related documents do show up if I reference an id with `/` (like, `products/1-A`), but it doesn't work with `products-1-A` .

Any idea if this is a bug or am I doing something wrong? Thank you in advance!

**Comment by ayende** [comment]
In the studio, we use the regex `externalIdRegex = /^\w+\/\w+/ig;`  to detect potential ids. 
This is a studio only behavior, so the correctness isn't mandated. Features like include, on the other hand, don't care about the pattern.

**Reply by MikeMoolenaar99** [reply]
Thank you for the quick reply, any plans to somehow edit that regex in any way? Maybe a setting in the Studio Configuration tab would be nice to change the seperator.

**Reply by ayende** [reply]
Not at this time, no

**Comment by danielwinkler** [comment]
Hi @MikeMoolenaar99 

I started working on an extension to do what you want, it is just a quick hack but i had no prior experience to doing chrome extensions
https://github.com/danielwinkler/raven-studio-chrome-extension

I have a hard coded regex in there `var regex = /"\w+[\/-]\d+-\w"/ig;` 
It is listening to navigation using the background script and also to scroll in the editor;

In the extension i do not have access to the JS of the webpage, so i decided to parse what is visible in the editor only;

You can download/clone the code and load it as extension in chrome://extensions by enabling local extensions

I put minimal effort into it; if you want to make it nicer feel free to improve it :)

**Reply by MikeMoolenaar99** [reply]
Hey, thank you so much for making this! I haven't tried it yet, but I will take a look soon and maybe make a PR. Maybe we can also release it to the Chrome and Firefox extension stores if it works properly.

**Reply by danielwinkler** [reply]
Sounds great!
